src/main/scala/strawman/collection/immutable/NumericRange.scala

```diff
@@ -80,6 +82,8 @@ final class NumericRange[T](
     else locationAfterN(idx)
   }
 
+  def updated[B >: T](index: Int, elem: B): IndexedSeq[B] = collection.generic.Updated.toImmutableArray(index, elem, coll)
```

Comment:
This could be implemented in a lazy way without materializing the Range. I find it counter-intuitive that it does that. We should either make it lazy or add a warning in the scaladoc.